### PR TITLE
Add `contract alias ls`.

### PR DESCRIPTION
**Pull request by fnando**
### What

```console
$ cargo run contract alias ls --global
   Compiling soroban-cli v21.4.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/soroban-cli)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 3.08s
     Running `target/debug/soroban contract alias list --global`
ℹ️ Aliases available for network 'Test SDF Network ; September 2015'
hello_world: CAXUMKGXAOY6IAO36XJV6DSOQBMBANSUJ6NR5EXIF27ZWNPW7OOTTHPC
increment: CALLOOCCCJRZ6AZAJSKNCTDZ766N4ANVRKRJPHC65KDV7RJCS37PS4XY

ℹ️ Aliases available for network 'Standalone Network ; February 2017'
hello_world: CC47LWR7ZJH2QQTFUUNRERMBPBAUHIVLKHIKACV6VVZEEQJIHJOJNGPL


$ echo $?
0

$ cargo run contract alias ls
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.20s
     Running `target/debug/soroban contract alias list`
⚠️ No aliases defined for network

$ echo $?
1
```

### Why

#1420 

### Known limitations

N/A

### CODE REVIEW

File: cmd/soroban-cli/src/commands/contract/alias/list.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        #[allow(unused_variables)]
+        let network_passphrase = &network.network_passphrase;
+        let config_dir = self.config_locator.config_dir()?;
+        let pattern = config_dir
+            .join("contract-ids")
+            .join("*.json")
+            .to_string_lossy()
+            .into_owned();
+
+        let paths = glob::glob(&pattern)?;
```
Comment (Ifropc) [inline cmd/soroban-cli/src/commands/contract/alias/list.rs:55]: If refactoring slightly we don't need to add a new dependency :)
```suggestion
        let paths = fs::read_dir(config_dir.join("contract-ids"))?
            .map(|res| res.map(|e| e.path()))
            .collect::<Result<Vec<_>, io::Error>>()?
            .into_iter()
            .filter(|e| e.extension() == Some("json".as_ref()));
```

File: cmd/soroban-cli/src/commands/contract/alias/list.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Locator(#[from] locator::Error),
+
+    #[error(transparent)]
+    Network(#[from] network::Error),
+
+    #[error("no contract found with alias '{alias}' for network '{network_passphrase}'")]
+    NoContract {
```
Comment (Ifropc) [inline cmd/soroban-cli/src/commands/contract/alias/list.rs:29]: Seems to be unused 🤔

File: FULL_HELP_DOCS.md [hunk 3]
```diff
@@ -163,6 +163,7 @@ Utilities to manage contract aliases
 * `remove` — Remove contract alias
 * `add` — Add contract alias
 * `show` — Show the contract id associated with a given alias
+* `list` — List all stored aliases for a given network
```
Comment (Ifropc) [inline FULL_HELP_DOCS.md:166]: Should it be `ls` instead? Also maybe `remove` -> `rm`?
To use the same style as `stellar keys`/`stellar network`

File: cmd/soroban-cli/src/commands/contract/alias/list.rs (hunk 1 above)
Comment (fnando) [reply cmd/soroban-cli/src/commands/contract/alias/list.rs:55]: The dependency was already there (by another crate). I just made it explicit. :)

File: FULL_HELP_DOCS.md (hunk 3 above)
Comment (fnando) [reply FULL_HELP_DOCS.md:166]: Good catch. I think so. I'll rename it in a separate pr.

**Review by willemneal — COMMENTED** [review]
My one thought is that it would be good to have a way to list all across all networks. Like -l on other list commands.

**Comment by fnando** [comment]
@willemneal I think that's a good idea (#1420 even mentions it), but I wonder how we can achieve that by using `command(flatten)`. 🤔

I tried to make it work, but failed to make everything work.

**Comment by Ifropc** [comment]
As an idea we can add list `--all` that conflicts with `--network`
The only thing is that our current flags are already pretty complicated, not sure if it makes sense to complicate it even further

**Comment by fnando** [comment]
@Ifropc I did that, but network options are required, and simply using `Option<network::Args>` doesn't work (it complains about `#[group(skip)]`, and even without it doesn't work).

**Comment by Ifropc** [comment]
Yeah, I was looking into groups, unfortunately we can't make it work easily while using derive API, this is a known issue in clap (and I don't think it will be changed, based on the discussion in Clap repo). So our only option is to migrate to builder for networks and maybe other flags as well, which we probably don't want to spend our time on right now.  
What I suggest is to not use `Option<network::Args>` and instead only use network name as a string (testnet/mainnet/etc) -> as we don't need to know RPC url to fetch aliases.
But it will conflict with how we are using network everywhere else, so maybe not a very good idea

**Comment by fnando** [comment]
I guess we should ignore `--all` for now.

**Comment by leighmcculloch** [comment]
I think we can keep this command simpler and output all aliases for all networks, and provide no filtering options.

**Comment by fnando** [comment]
@leighmcculloch works for me!

**Comment by fnando** [comment]
The list output is currently this:

```console
$ cargo run contract alias list --global
   Compiling soroban-cli v21.4.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/soroban-cli)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 3.08s
     Running `target/debug/soroban contract alias list --global`
ℹ️ Aliases available for network 'Test SDF Network ; September 2015'
hello_world: CAXUMKGXAOY6IAO36XJV6DSOQBMBANSUJ6NR5EXIF27ZWNPW7OOTTHPC
increment: CALLOOCCCJRZ6AZAJSKNCTDZ766N4ANVRKRJPHC65KDV7RJCS37PS4XY

ℹ️ Aliases available for network 'Standalone Network ; February 2017'
hello_world: CC47LWR7ZJH2QQTFUUNRERMBPBAUHIVLKHIKACV6VVZEEQJIHJOJNGPL


$ echo $?
0

$ cargo run contract alias list
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.20s
     Running `target/debug/soroban contract alias list`
⚠️ No aliases defined for network

$ echo $?
1
```

**Review by chadoh — APPROVED** [review]
Thanks!

**Pull request merged**